pkg/virt-operator/install-strategy/reconcile.go

```diff
@@ -0,0 +1,44 @@
+package installstrategy
+
+import (
+	"reflect"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"kubevirt.io/controller-lifecycle-operator-sdk/pkg/sdk"
+)
+
+const LastAppliedConfigAnnotationKey = "kubevirt.io/last-applied-configuration"
+
+func setLastAppliedConfiguration(obj metav1.Object) error {
+	return sdk.SetLastAppliedConfiguration(obj, LastAppliedConfigAnnotationKey)
+}
+
+func resourceShouldUpdate(currentRuntimeObj, desiredRuntimeObj runtime.Object) (bool, runtime.Object, error) {
+	currentMetaObj := currentRuntimeObj.(metav1.Object)
+	desiredMetaObj := desiredRuntimeObj.(metav1.Object)
+
+	currentRuntimeObj, err := sdk.StripStatusFromObject(currentRuntimeObj)
+	if err != nil {
+		return false, currentRuntimeObj, err
+	}
+
+	currentRuntimeObjCopy := currentRuntimeObj.DeepCopyObject()
+	// allow users to add new annotations (but not change ours)
+	sdk.MergeLabelsAndAnnotations(desiredMetaObj, currentMetaObj)
+
+	if !sdk.IsMutable(currentRuntimeObj) {
+		err = setLastAppliedConfiguration(desiredMetaObj)
+		if err != nil {
+			return false, currentRuntimeObj, err
+		}
+	}
+
+	currentRuntimeObj, err = sdk.MergeObject(desiredRuntimeObj, currentRuntimeObj, LastAppliedConfigAnnotationKey)
```

Comment:
Let's investigate https://github.com/openshift/library-go/tree/master/pkg/operator/resource/resourceapply closely and follow their example. It can cope with defaulting and we can simply do an update without merging, since we upfront really find out if we have to update the resource.

Reply:
What exactly do you mean by "cope with defaulting"?

Reply:
post an object "like a deployment", mutating webhook injects some defaults into the object, resulting object's spec doesn't reflect.DeepEquals the original one posted.  If we're strictly reconciling objects, we had to account for that defaulting or we'll get into a hotloop trying to correct the defaults.